### PR TITLE
fix(Scripts/World: Spawn adds should scale with raid size

### PR DESCRIPTION
**Pull request by Maelthyrr**
## Changes Proposed:
-  Spawn adds will scale with raid size. Max is set to 15 adds.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/12279

## SOURCE:

Cherry-pick: https://github.com/vmangos/core/blob/development/src/scripts/world/dragons_of_nightmare/boss_ysondre.cpp

## Tests Performed:

- None

## How to Test the Changes:

1. .go c id 14887
2. engage Ysondre with different number of raiders
3. .damage 90000
4. Count the number of adds

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Maelthyrr** [comment]
@UltraNix can I get your review ? I'm really not sure about the line 276, but I don't find another way to use the condition isAlive and isGamemaster.

**Review by acidmanifesto — APPROVED** [review]

File: src/server/scripts/World/boss_emerald_dragons.cpp [hunk 1]
```diff
@@ -268,7 +268,20 @@ class boss_ysondre : public CreatureScript
             {
                 Talk(SAY_YSONDRE_SUMMON_DRUIDS);
 
-                for (uint8 i = 0; i < 10; ++i)
+                auto const& attackers = me->GetThreatMgr().getThreatList();
+                uint8 attackersCount = 0;
+
+                for (const auto attacker : attackers)
+                {
+                    if (Player* pPlayer = attacker->GetSource()->GetOwner()->ToPlayer())
```
Comment (UltraNix) [inline src/server/scripts/World/boss_emerald_dragons.cpp:276]: `Player* player` Say `No` to hungarian notations

File: src/server/scripts/World/boss_emerald_dragons.cpp [hunk 2]
```diff
@@ -268,7 +268,20 @@ class boss_ysondre : public CreatureScript
...
+                        if (pPlayer && pPlayer->IsAlive() && !pPlayer->IsGameMaster())
```
Comment (UltraNix) [inline src/server/scripts/World/boss_emerald_dragons.cpp:277]: `pPlayer` check needed - done above.

**Review by UltraNix — CHANGES_REQUESTED** [review]
`if (Player* pPlayer = attacker->GetSource()->GetOwner()->ToPlayer())`
`attacker->GetSource()->GetOwner()` Are you sure it will never crash? `GetOwner` is always there?

**Comment by Maelthyrr** [comment]
> `if (Player* pPlayer = attacker->GetSource()->GetOwner()->ToPlayer())` `attacker->GetSource()->GetOwner()` Are you sure it will never crash? `GetOwner` is always there?

It's was I asked, I don't see what use using the threatlist to check the state of players. Maybe isOnline ?

**Comment by Nyeriah** [comment]
Just one consideration: game masters (gm mode on) should never be in a creature's threat list as well as dead players (but I'm not entirely sure about the second ones)

**Comment by Maelthyrr** [comment]
I ask myself. If a player is in threatlist, he should be alive? If he die, he gone. So the two conditions are just useless ?

**Comment by Nyeriah** [comment]
The GM one yes I'm sure, they're not valid targets so they can't be added to the threatlist. As for the dead players, I am not entirely sure, but I think we don't store references for dead players, yes

**Comment by temperrr** [comment]
- [ ] 1 raider - 3 adds
- [ ] 5 raiders - 3 adds

**Comment by Gultask** [comment]
I had the same results as temperrr but I tested it before the latest updates. I'll check again.

Also, I think you can remove the minimum spawn of 3 and set it to 1. In WotLK there wasn't a minimum of 3.
https://youtu.be/cheP7oKQdN4?t=39 here spawns only one.

**Comment by Maelthyrr** [comment]
Need to add a check to be sure that a player is target. But I don't how treat with threatref to apply these checks. So I let someone else fix this.

**Comment by temperrr** [comment]
tested, works

**Pull request closed without merging**